ACadSharp/Entities/MText.Format.cs

```diff
@@ -0,0 +1,302 @@
+			/// </summary>
+			/// <param name="formats">Format string to use.</param>
+			internal Format(string formats)
+			{
+				// Used only for testing
+				if (formats.Contains("L")) this.IsUnderline = true;
+
+				if (formats.Contains("O")) this.IsOverline = true;
+
+				if (formats.Contains("K")) this.IsStrikeThrough = true;
+
+				if (formats.Contains("l")) this.IsUnderline = false;
+
+				if (formats.Contains("o")) this.IsOverline = false;
+
+				if (formats.Contains("k")) this.IsStrikeThrough = false;
+			}
+
+			/// <summary>
+			/// Overrides the current format values with the passed format
+			/// </summary>
+			/// <param name="source">Source format to copy from.</param>
+			public void OverrideFrom(Format? source)
+			{
+				if (source == null)
+					return;
+
+				this.IsUnderline = source.IsUnderline;
+				this.IsOverline = source.IsOverline;
+				this.IsStrikeThrough = source.IsStrikeThrough;
+				this.Font.OverrideFrom(source.Font);
+				this.Height = source.Height;
+				this.IsHeightRelative = source.IsHeightRelative;
+				this.Width = source.Width;
+				this.Obliquing = source.Obliquing;
+				this.Tracking = source.Tracking;
+				this.Align = source.Align;
+				this.Color = source.Color;
+				this.Paragraph.Clear();
+				this.Paragraph.AddRange(source.Paragraph);
+			}
+
+			/// <summary>
+			/// Resets the format to the default values for reuse.
+			/// </summary>
+			public void Reset()
+			{
+				this.IsUnderline = false;
+				this.IsOverline = false;
+				this.IsStrikeThrough = false;
+				this.Font.Reset();
+				this.Height = null;
+				this.IsHeightRelative = false;
+				this.Width = null;
+				this.Obliquing = null;
+				this.Tracking = null;
+				this.Align = null;
+				this.Color = null;
+				this.Paragraph.Clear();
+			}
+
+			public override bool Equals(object? obj)
+			{
+				return this.Equals(obj as Format);
+			}
+
+			/// <summary>
+			/// Checks to see if the passed format equals this format.
+			/// </summary>
+			/// <param name="other">Other format to compare to.</param>
+			/// <returns>True of the formats are equal, false otherwise.</returns>
+			public bool Equals(Format? other)
+			{
+				if (other == null)
+					return false;
+
+				return this.IsUnderline == other.IsUnderline
+				       && this.IsOverline == other.IsOverline
+				       && this.IsStrikeThrough == other.IsStrikeThrough
+				       && isApproximateEqual(this.Height, other.Height)
+				       && this.IsHeightRelative == other.IsHeightRelative
+				       && isApproximateEqual(this.Width, other.Width)
+				       && isApproximateEqual(this.Obliquing, other.Obliquing)
+				       && isApproximateEqual(this.Tracking, other.Tracking)
+				       && Nullable.Equals(this.Align, other.Align)
+				       && this.Color.Equals(other.Color)
+				       && this.Font.Equals(other.Font)
+				       && this.AreParagraphsEqual(other.Paragraph);
+			}
+
+			[SuppressMessage("ReSharper", "NonReadonlyMemberInGetHashCode")]
```

Comment:
Can this attribute be removed? it's only to keep the code cleaner, if is a headache for development it can stay.

Reply:
Done.  See https://github.com/DomCR/ACadSharp/pull/94#discussion_r1071359426